src/lib/storage/base_frame_of_reference_segment.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <memory>
+
+#include "base_encoded_segment.hpp"
+
+namespace opossum {
+
+class BaseCompressedVector;
+
+/**
+ * @brief Base class of FrameOfReferenceSegment<T> exposing type-independent interface
+ */
+
+class BaseFrameOfReferenceSegment : public BaseEncodedSegment {

```

Comment:
Is this class really necessary?

Reply:
No. FrameOfReferenceSegment directly inherits from BaseEncodedSegment now. This file should not exist anymore. Will delete it